src/exodus/sigmadb.cpp

```diff
@@ -298,13 +296,33 @@ void CMPMintList::RecordGroupSize(uint16_t groupSize)
 {
     auto key = CreateGroupSizeKey();
 
-    exodus::swapByteOrder(groupSize);
     auto status = pdb->Put(writeoptions, GetSlice(key),
         leveldb::Slice(reinterpret_cast<char*>(&groupSize), sizeof(groupSize)));
 
     if (!status.ok()) {
-        LogPrintf("%s: Store sigma mint group size fail\n", __func__);
+        throw std::runtime_error("store sigma mint group size fail");
+    }
+}
+
+uint16_t CMPMintList::GetGroupSize()
+{
+    auto key = CreateGroupSizeKey();
+
+    std::string result;
+    auto status = pdb->Get(readoptions, GetSlice(key), &result);
+
+    if (status.ok()) {
+        uint16_t groupSize(0);
+
+        if (result.size() == sizeof(groupSize)) {
+            std::copy_n(result.data(), result.size(), reinterpret_cast<char*>(&groupSize));
+            return groupSize;
+        }
+
+        throw std::runtime_error("size of group size value is invalid");
     }
+
+    return 0;
```

Comment:
We should return `0` only when the key is not found. In other cases we should throw.